### PR TITLE
feat: add SectionCard component

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- Adds `<SectionCard />` component
- _Responsive behavior is TBD later_

## ✅ Checklist

Easy UI has certain UX standards that must be met. In general, non-trivial changes should meet the following criteria:

- [x] Visuals match Design Specs in Figma
- [x] Stories accompany any component changes
- [x] Code is in accordance with our style guide
- [x] Design tokens are utilized
- [x] Unit tests accompany any component changes
- [x] TSDoc is written for any API surface area
- [x] Specs are up-to-date
- [x] Console is free from warnings
- [x] No accessibility violations are reported
- [x] Cross-browser check is performed (Chrome, Safari, Firefox)
- [x] Changeset is added

~Strikethrough~ any items that are not applicable to this pull request.

### CODE REVIEW

File: easy-ui-react/src/SectionCard/SectionCard.module.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+.header {
+  display: flex;
+  flex-wrap: nowrap;
+  align-items: center;
+  justify-content: space-between;
+  min-width: 0;
+
+  // Ensure middle content is centered
+  > :nth-child(1),
+  > :nth-child(3) {
```
Comment (OskiTheCoder) [inline easy-ui-react/src/SectionCard/SectionCard.module.scss:13]: **question**: is there a mockup that indicates a `HorizontalStack` wouldn't work for `SectionCard.Header`?

File: easy-ui-react/src/Card/Card.tsx [hunk 2]
```diff
@@ -103,6 +104,20 @@ export type CardAreaProps = {
...
+export function pickCardAreaProps(props: Partial<CardAreaProps>) {
+  return pick(props, [
+    "background",
+    "children",
+    "padding",
+    "paddingX",
+    "paddingY",
+    "as",
+  ]);
+}
```
Comment (OskiTheCoder) [inline easy-ui-react/src/Card/Card.tsx:119]: nice

**Review by OskiTheCoder — APPROVED** [review]
LGTM! no blocking concerns

File: easy-ui-react/src/SectionCard/SectionCard.module.scss (hunk 1 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/SectionCard/SectionCard.module.scss:13]: `HorizontalStack` doesn't work well currently for true-centering the middle column. i'm going to look into a better solution for these cases

**Pull request merged**